webui/src/lib/components/repository/GeoJSONPreview.jsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { MapContainer, TileLayer, GeoJSON } from "react-leaflet";
+import L from 'leaflet';
+
+const parseGeoJSON = data => {
+    try {
+        return JSON.parse(data);
+    } catch {
+        return null;
+    }
+};
+
+export const GeoJSONPreview = ({ data }) => {
+    if (!data) return <div>No data to preview</div>;
+
+    const geoJsonData = parseGeoJSON(data);
+    if (!geoJsonData) return <div>Invalid GeoJSON data</div>;
+
+    const bounds = L.geoJSON(geoJsonData).getBounds();
+
+    const mapProps = bounds.isValid()
+        ? { bounds }
+        : { center: [0, 0], zoom: 2 };
+
+    return (
+        <div style={{ height: "400px", width: "100%" }}>
```

Comment:
We should try to avoid inline styles like this as much as possible.
I suggest adding a class to `globals.css` and using it here (and also for `<MapContainer>`).